server/src/main/java/com/linecorp/centraldogma/server/internal/storage/DirectoryBasedStorageManager.java

```diff
@@ -347,11 +419,51 @@ private static boolean isValidChildName(String name) {
             return false;
         }
 
-        return !name.endsWith(SUFFIX_REMOVED);
+        return !name.endsWith(SUFFIX_REMOVED) && !name.endsWith(SUFFIX_PURGED);
     }
 
     @Override
     public String toString() {
         return Util.simpleTypeName(getClass()) + '(' + rootDir + ')';
     }
+
+    private final class StorageRemovalManager {
+
+        private static final String REMOVAL_TIMESTAMP_NAME = ".removal.timestamp";
```

Comment:
I think we do not need to make this a hidden file. How about just `removal.timestamp`?

Reply:
Thanks, I will change this into a regular file. :-)